File: src/main/java/us/n8l/duplicatefinder/FileEntry.java
package us.n8l.duplicatefinder;

import org.jetbrains.annotations.Nullable;
import org.joda.time.DateTime;
import us.n8l.duplicatefinder.digests.MD5;
import us.n8l.duplicatefinder.digests.SHA256;

import java.util.Optional;

public class FileEntry {
    long id;
    long size; // in bytes
    MD5 headerHash;
    SHA256 hash;
    DateTime mtime;
    DateTime ctime;
    DateTime atime;
}

File: src/main/java/us/n8l/duplicatefinder/FileEntry.kt
package us.n8l.duplicatefinder

import us.n8l.duplicatefinder.digests.MD5
import us.n8l.duplicatefinder.digests.SHA256
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.attribute.BasicFileAttributes
import java.nio.file.attribute.FileTime
import java.util.*
import java.util.concurrent.atomic.AtomicReference


